packages/performance-tests/test/TransformerRegression.test.ts

```diff
@@ -102,17 +107,60 @@ async function runRegressionTests() {
   describe("Transformer Regression Tests", function () {
     testIModels.forEach(async (iModel) => {
       describe(`Transforms of ${iModel.name}`, async () => {
+        let sourceDb: BriefcaseDb;
+        let reportInfo: ReporterInfo
+
+        before(async () => {
+          Logger.logInfo(loggerCategory, `processing iModel '${iModel.name}' of size '${iModel.tShirtSize.toUpperCase()}'`);
+          sourceDb = await iModel.load();
+          const fedGuidSaturation = sourceDb.withStatement(
+            `
+            SELECT
+            CAST(SUM(hasGuid) as DOUBLE)/SUM(total) ratio 
+            FROM (
+              SELECT IIF(FederationGuid IS NOT NULL, 1, 0) hasGuid,
+              1 as total FROM bis.Element
+            )`,
+            (stmt) => {
+              assert(stmt.step() === DbResult.BE_SQLITE_ROW);;
+              return stmt.getValue(0).getDouble();
+            }
+          );
+          Logger.logInfo(loggerCategory, `Federation Guid Saturation '${fedGuidSaturation}'`);
+          const toGb = (bytes: number) => `${(bytes / 1024 **3).toFixed(2)}Gb`;
+          const sizeInGb = toGb(fs.statSync(sourceDb.pathName).size);
+          Logger.logInfo(loggerCategory, `loaded (${sizeInGb})'`);
+          reportInfo = {
+            "Id": iModel.iModelId,
+            "T-shirt size": iModel.tShirtSize,
+            "Gb size": sizeInGb,
+            "Branch Name": branchName,
+            "Federation Guid Saturation 0-1": fedGuidSaturation,
+          };
+        });
+
         testCasesMap.forEach(async (testCase, key) => {
+          before(async () => {
+            if(!sourceDb.isOpen){
+              sourceDb = await BriefcaseDb.open({
+                fileName: sourceDb.pathName,
+                readonly: true,
+              });
+            }
+          });
+
           it(key, async () => {
-            reporter = await testCase(iModel, reporter, branchName);
+            reporter = await testCase(sourceDb, reporter, reportInfo);
           }).timeout(0);
+          
+          sourceDb.close();
```

Comment:
this can execute before the async callback passed to `it(key, ...)` is called. You should close the sourceDb inside the async callback body of `it(key...)`

Reply:
this seemed to cause the same error when trying to use an imodel that is closed, but that may be due to issues with the if stmt inside the before, so hopefully fixing that will also fix this.

Reply:
reasonable.